### ISSUE TITLE
Support for checking if the window has focus

### ISSUE THREAD

**Issue by unclezs** [wailsapp/wails#2896]
### Is your feature request related to a problem? Please describe.

I want to implement using a keyboard shortcut to hide/show the application.

### Describe the solution you'd like

Support for checking if the window has focus

**Comment by tellixu** [comment]
Strongly support this suggestion

**Comment by leaanthony** [comment]
Does `application.Hide` not do what you want though?

**Comment by unclezs** [comment]
> Does `application.Hide` not do what you want though?

Yes, sometimes the window is merely obscured by other application windows rather than hidden. In such cases, triggering the shortcut key should bring the window to the front, but currently, I have no way of determining whether the window is obscured. Users cannot definitively decide whether to call the 'hide' or 'show' method when they trigger the shortcut key.

**Comment by EFHDev** [comment]
I would also very much like this! 👍

**Comment by leaanthony** [comment]
This is available in v3. I would be happy to merge a PR back-porting this to v2. I will close this for now but feel free to link it in the PR if you raise one 👍
